packages/app/src/admin/SuperAdminPage.tsx

```diff
@@ -207,34 +207,86 @@ export function SuperAdminPage(): JSX.Element {
   );
 }
 
-function startAsyncJob(medplum: MedplumClient, title: string, url: string, body?: Record<string, string>): void {
+type AsyncJobOptions = {
+  body?: Record<string, string>;
+  cancellable?: boolean;
+};
+
+function startAsyncJob(medplum: MedplumClient, title: string, url: string, asyncJobOptions?: AsyncJobOptions): void {
+  const controller = new AbortController();
   notifications.show({
     id: url,
     loading: true,
     title,
     message: 'Running...',
     autoClose: false,
-    withCloseButton: false,
+    ...(asyncJobOptions?.cancellable
+      ? { withCloseButton: true, onClose: () => controller.abort() }
+      : { withCloseButton: false }),
   });
 
-  const options: MedplumRequestOptions = { method: 'POST', pollStatusOnAccepted: true };
-  if (body) {
-    options.body = JSON.stringify(body);
+  const options: MedplumRequestOptions = {
+    method: 'POST',
+    pollStatusOnAccepted: true,
+    ...(asyncJobOptions?.cancellable ? { asyncReqCancelSignal: controller.signal } : undefined),
+  };
+
+  if (asyncJobOptions?.body) {
+    options.body = JSON.stringify(asyncJobOptions.body);
   }
 
   medplum
-    .startAsyncRequest(url, options)
-    .then(() => {
-      notifications.update({
-        id: url,
-        color: 'green',
-        title,
-        message: 'Done',
-        icon: <IconCheck size="1rem" />,
-        loading: false,
-        autoClose: false,
-        withCloseButton: true,
-      });
+    .startAsyncRequest<AsyncJob>(url, options)
+    .then((job) => {
+      switch (job.status) {
+        case 'completed':
+          notifications.update({
+            id: url,
+            color: 'green',
+            title,
+            message: 'Done',
+            icon: <IconCheck size="1rem" />,
+            loading: false,
+            autoClose: false,
+            withCloseButton: true,
+          });
+          break;
+        case 'cancelled': {
+          const cancelledNotif = {
+            id: url,
+            color: 'red',
+            title,
+            message: 'Job cancelled',
+            icon: <IconX size="1rem" />,
+            loading: false,
+            autoClose: false,
+            withCloseButton: true,
+          };
+          // If the signal has been aborted, it means we made the change via our X button
+          // That removes the notification so we have to show it again
+          if (controller.signal.aborted) {
```

Comment:
Hm.. I thought `update` would bring back the notification if closed, no?

Reply:
Nope, it does not. It stays closed sadly